### DISCUSSION TITLE
Questions about cross validation on phylosem

### DISCUSSION THREAD

**Discussion by mamouud** [James-Thorson-NOAA/phylosem#12] [Q&A]
Hi James et al.
I'm using  Phylosem package for imputing missing data within my trait databases.
In an effort to verify the quality of my  imputed data, I aimed to employ cross-validation, similar to your approach. However, the task
became more challenging due to the absence of a 'predict' function in the Phylosem package. While attempting to follow your methodology, I found myself a bit confused. Your paper mentions that you obtain parameters from the model trained on three-quarters of the data and subsequently use them to apply the model on the remaining quarter.
Could you please provide some insights on how you precisely manage the reuse of the trained model's parameters?
Are these parameters directly utilized in the SEM equations related to each trait? 
If so, does this mean that the phylogenetic information is captured within these parameters?
Thanks very much!

**Comment by James-Thorson-NOAA** [comment]
Thanks for posting the question in a searchable way :)

The easiest way to do this is to take dataset `data`, identify your training and testing partitions, and replace the testing partition with NA values while retaining all rows in `data`.  Then refit the model, convert to phylo4d using `as_phylo4d`, extract the imputed values for the testing partition (which were NAs), and compare those with the withheld values.  Then obviously replicate as needed.

Closing for now, but happy to discuss here.
